java/java.editor/src/org/netbeans/modules/java/editor/codegen/ToStringGenerator.java

```diff
@@ -303,4 +321,10 @@ private static MethodInvocationTree createAppendInvocation(TreeMaker make, Expre
         );
     }
 
+    private static ExpressionTree makeExpression(TreeMaker make, Element element) {
+        return element.getKind() == ElementKind.METHOD
+                ? make.MethodInvocation(Collections.emptyList(), make.MemberSelect(make.Identifier("this"), element), Collections.emptyList()) // NOI18N
+                : make.Identifier(element.getSimpleName());
+    }
```

Comment:
@lahodaj here I didn't know how to do a method invocation on implicit "this". So i cheated a bit after I saw that the doc says that `Identifier("this")` would do no validation.

Reply:
If you mean doing `componentName()`, then `make.MethodInvocation(Collections.emptyList(), make.Identifier(element.getSimpleName()), Collections.emptyList())` should work, I think.

Reply:
oh wow that is easier. Don't know how I overlooked it. Working on an update if the thunderstorm allows it.